docsk8s/quick-start/sample-configuration-files.asciidoc

```diff
@@ -33,62 +45,245 @@ This contains the Logstash settings and pipeline configuration:
+<11> The {es} password is taken from `demo-es-elastic-user` and passed to the Logstash pipeline as an `ELASTICSEARCH_PASSWORD` environment variable. Refer to link:https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-request-elasticsearch-endpoint.html[Access the {es} endpoint] in the {eck} Guide for details.
 
+[[qs-service]]
 `003-service.yaml`::
-Sets the TCP ports for the Filebeat and Metricbeat services.
++
+This file contains the Service definition, opening up ports on the logstash pods to the internal metricbeat (for stack monitoring) and filebeat in this instance
+
+[source,yaml]
+--
+spec:
+  type: ClusterIP
+  ports:
+    - port: 9600 <1>
+      name: "stats"
+      protocol: TCP
+      targetPort: 9600 <1>
+    - port: 5044 <2>
+      name: "beats"
+      protocol: TCP
+      targetPort: 5044 <2>
+  selector:
+    app: logstash-demo
+--
 
-`004-hpa.yaml`::
-The Horizontal Pod Autoscaler is used to configure the horizontal scaling details for CPU and memory for the {ls} instance.
+<1> Opening up port `9600` for metricbeat to connect to the Logstash metrics API
+<2> Opening up port `5044` for filebeat to connect to the beats input defined in the <<qs-configmap,ConfigMap>>
+
+[[qs-additional-logstash-configuration]]
+
+[[qs-autoscaler]]
+`004-hpa.yml`::
 
-`005-filebeat.yaml`::
-Includes the mount path for the generated certificate and key files, the secrets file `logstash-beats-tls`, and a {ls} output configuration as follows:
 +
+This sets up a horizontal pod autoscaler to scale logstash instances up and down, depending on the load on the logtash instance(s). See link:https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/[kubernetes autoscaler docs] for more details.
+
 [source,yaml]
 --
-    output.logstash:
-      hosts:
-        - "logstash:5044"
-      ssl.certificate_authorities: ["/usr/share/filebeat/ca.crt"]
-      ssl.certificate: "/usr/share/filebeat/client.crt"
-      ssl.key: "/usr/share/filebeat/client.key"
+apiVersion: autoscaling/v2 <1>
+kind: HorizontalPodAutoscaler
+metadata:
+  name: logstash
+  labels:
+    app: logstash-demo
+spec:
+  minReplicas: 1 <2>
+  maxReplicas: 2
+  behavior:
+    scaleUp:
+      stabilizationWindowSeconds: 60 <3>
+    scaleDown:
+      stabilizationWindowSeconds: 180
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: logstash <4>
+  metrics:
+    - type: Resource <5>
+      resource:
+        name: cpu
+        target:
+          type: Utilization
+          averageUtilization: 80
+    - type: Resource
+      resource:
+        name: memory
+        target:
+          type: Utilization
+          averageUtilization: 80
 --
+<1> Requires kubernetes `1.23` and up
+<2> Specify the maximum and minimum number of Logstashes desired for the cluster
+<3> Specify stabilization windows to avoid "flapping" the replica count, using a rolling
+<4> This refers to the `Deployment` created <<qs-deployment, earlier>>
```

Comment:
```suggestion
<1> Requires {k8s} `1.23` and higher.
<2> Specifies the maximum and minimum number of Logstashes desired for the cluster.
<3> Specifies stabilization windows to avoid "flapping" the replica count, using a rolling
<4> `Deployment` created <<qs-deployment, earlier>>
```